### PR TITLE
Bug 1414582 - Print warning for "sample"

### PR DESCRIPTION
**Pull request by relud**
Fixes Bug 1414582 Print a warning when using the "sample" functionality of Dataset

r? @mreid-moz

### CODE REVIEW

**Comment by mreid-moz** [comment]
Looks good to me, r? @sunahsuh for the copy of the warning.

File: moztelemetry/dataset.py [hunk 1]
```diff
@@ -253,6 +253,11 @@ def records(self, sc, limit=None, sample=1, seed=42, decode=None, summaries=None
         if summaries and limit is None and sample != 1:
             if sample < 0 or sample > 1:
                 raise ValueError('sample must be between 0 and 1')
+            print(
+                "WARNING: sampling based on the set of files in s3. This\n"
```
Comment (sunahsuh) [inline moztelemetry/dataset.py]: I think the first line should very explicitly be: this is not a representative sample, and we should be explicit about the implications e.g.

 "WARNING: THIS IS NOT A REPRESENTATIVE SAMPLE. This "sampling" is based on s3 files and is highly susceptible to skew. Use only for quicker performance while prototyping."

**Comment by relud** [comment]
okay, updated the warning

**Comment by sunahsuh** [comment]
👍 :shipit:

**Pull request merged**